### ISSUE TITLE
Copilot Won't Work

### ISSUE THREAD

**Issue by eric-bodhi** [eric-bodhi/configs#2]
Only way to get CoPilot working is to change `.local/share/nvim/lazy/nvim-lspconfig/lua/lspconfig/server_configurations/clangd.lua`
```
local cmp_nvim_lsp = require "cmp_nvim_lsp"

require("lspconfig").clangd.setup {
  on_attach = on_attach,
  capabilities = cmp_nvim_lsp.default_capabilities(),
  cmd = {
    "clangd",
    "--offset-encoding=utf-16",
  },
}
```

However lazyvim will not let you change the config and will fail the plugin:
![Screenshot from 2024-01-21 21-59-43](https://github.com/eric-bodhi/configs/assets/101071809/84fdd8aa-738b-47a9-a71f-6e6ae01e505f)

Warning that continously pops up is:
" multiple different client offset_encodings detected for buffer, this is not supported yet"

**Comment by eric-bodhi** [comment]
"There's no reason to touch the lspconfig files. You can configure lspconfig per server from your user config folder just fine.

Have you tried the lazyvim clangd extra? If you don't want to just use it blindly, you can at least take a look at how it sets up clangd-specific settings:

https://lazyvim.github.io/extras/lang/clangd#nvim-lspconfig

Edit: also the plugin isn't failing - it should still work fine. it's just that trying to pull new updates will fail." - Reddit. Eric in the future, I'm going to bed rn, check the link, you can make the required modifications without touching lspconfigs all through lazyvim

**Comment by eric-bodhi** [comment]
Fixed with commit #44
